code/modules/food_and_drink/ingredients.dm

```diff
@@ -582,41 +582,45 @@
 	var/toppingstext = null
 
 	attackby(obj/item/W as obj, mob/user as mob)
-		if (istype(W, /obj/item/reagent_containers/food/snacks/))
-			var/obj/item/reagent_containers/food/snacks/F = W
-			if(!F.custom_food)
-				return
+		if (istype(W, /obj/item/reagent_containers/food))
+			add_ingredient(W, user)
+		else
+			return
+	proc/add_ingredient(obj/item/W as obj, mob/user as mob)
+		var/obj/item/reagent_containers/food/snacks/F = W
+		if(!F.custom_food)
+			return
+		if(user)
 			boutput(user, "<span class='notice'>You add [W] to [src].</span>")
-			topping = 1
+		topping = 1
+		if(istype(F,/obj/item/reagent_containers/food/snacks))
```

Comment:
`F` is either `null` (if `W` is not a `/obj/item/reagent_containers/food/snacks`) or is one of these. You don't really need to type check so much as check existence.

Reply:
Actually untrue, I have a drinks list and this type check is to see if it's not a drink

Reply:
Okay so, I actually tried changing this like you suggest, but it does immediately runtime because somehow `/obj/item/reagent_containers/food/drinks/bottle/drowsy` was happily assigned to `F` despite not being a snack at all! One does wonder!